joomla/joomla-cms
# Datepicker with 12 hour time format midnight error

PR for #17938

Thanks to @gclw for the code and his patience

### Test instructions
in \administrator\components\com_content\models\forms\article.xml
add the following two lines to the "created" field

`timeformat="12"
format="%Y-%m-%d %H:%M:%S"
`
As you set the time in the calendar observe the time in the field. 
1. Set the time to 4pm and the field shows 16:00
2. Now set the time to 12am and the field will show 12:00

This is the bug. It should show 00:00

Apply this PR,clear cache, and it will behave correctly as in the video below

### after PR

![clock](https://user-images.githubusercontent.com/1296369/31204796-750f8dfa-a965-11e7-95f0-e5e94d1ac7de.gif)
